src/condition.cpp

```diff
@@ -202,8 +202,11 @@ Condition* Condition::createCondition(ConditionId_t id, ConditionType_t type, in
 		case CONDITION_SPELLGROUPCOOLDOWN:
 			return new ConditionSpellGroupCooldown(id, type, ticks, buff, subId, aggressive);
 
-		case CONDITION_INFIGHT:
 		case CONDITION_DRUNK:
+			param = (param <= 0 ? 25 : param);
```

Comment:
Why this?

Reply:
This is really interesting and I'll have to try it sometime ;)

Reply:
> Why this?

To ensure backwards compatibility with the LUA interface. For instance, think of the `data/actions/scripts/other/fluids.lua` file. It looks something like this:

```lua
local drunk = Condition(CONDITION_DRUNK)
drunk:setParameter(CONDITION_PARAM_TICKS, 60000)

...
```

The parameter `CONDITION_PARAM_DRUNKENNESS` is never set. So, when this call to the LUA interface reaches that point in the code, the `param` value is 0, which would not cause a `drunkenness` effect on the creature. Therefore, I added this sanity check in order to fall back to the default parameter before the changes, 25, in case this condition is met.

(I hope it makes sense :p)

Reply:
You can just set the default value in header file to 25 instead of checking it

Reply:
Are you sure?

I could set it to 25 in the header file, but when the `createCondition` method calls the `ConditionDrunk` constructor it will overwrite it with a 0 because the default `param` value is, indeed, 0.

```c++
ConditionDrunk(ConditionId_t id, ConditionType_t type, int32_t ticks, bool buff, uint32_t subId, uint32_t drunkenness, bool aggressive = false) :
	Condition(id, type, ticks, buff, subId, aggressive), drunkenness(drunkenness) {}
```

Reply:
Well, you can just move the assignment to contructor block and check if drunkenness is 0

Reply:
True. Let me change it real quick and the PR should be good to merge.

Thank you very much :p.